tests/test_nms.cpp

```diff
@@ -13,8 +14,6 @@ AnchorMat create_sample_input() {
   Anchor anchor_5{0.3, 1.6, 1.1, 0.9, 0.3};
   Anchor anchor_6{0.4, 1.7, 1.0, 1.0, 0.2};
```

Comment:
Please provide a documentation as to why the test cases where created how they were chosen and what they represent and do they completely test your function. You can write all of this as doxygen comments after each test case and explain what this tests. I need to know the underlying logic of why these testcases are enough to test the robustness of the function.

Reply:
Do I have to write a documentation about it?

Reply:
The purpose of unit tests is to write test cases that make sure that the function will not break or give errors or wrong answers no matter what input is given as the arguments to the function. So, I just want a short description of why you chose the numbers in the Anchors as your testcases are they representative enough to guarantee that the unit testing is complete. The short description helps me or somebody who works on your code later evaluate if you have though of the test cases from every angle and have not missed anything. 

A small example is below
```
int add_natural_numbers(int a, int b)
    return a + b
```

If a negative number is given then this should not add them but without the safety checks this testcase fails and the new code then becomes

```
int add_natural_numbers(int a, int b)
    if (a >0 && b > 0)
          return a + b
   else
          throws excception("only positive numbers excepted")
```